`jmwalletd` exits when maker cannot be started

Finally managed to reproduce situations where `jmwalletd` just exits.

Expected: When the maker cannot be started, jmwalletd keeps running (or exits with error).
Actual: When the maker cannot be started, jmwalletd exits (cleanly).

The following are the minimal steps to reproduce the behaviour (on network `regtest`): 
1. Wallet with 3 _frozen_ utxos
2. Start the maker via `POST /api/v1/wallet/funded.jmdat/maker/start`
3. Wait 10 minutes
4. `jmwalletd` exits cleanly

Here are the logs from `logs/jmwalletd_stdout.log`:
```
<Request at 0x7f6fd46fa130 method=POST uri=/api/v1/wallet/funded.jmdat/maker/start clientproto=HTTP/1.1>
b'POST'
b'/api/v1/wallet/funded.jmdat/maker/start'
{}
2022-03-27 00:05:35,307 [INFO]  JoinMarket 0.9.6dev
2022-03-27 00:05:35,319 [DEBUG]  connection was made, starting client.
2022-03-27 00:05:35,329 [DEBUG]  building irc
2022-03-27 00:05:35,330 [DEBUG]  build_irc:  irc 6667 #joinmarket-pit-test
2022-03-27 00:05:36,638 [DEBUG]  signedOn:  ergo.test
2022-03-27 00:05:37,308 [ERROR]  do not have any coins left
2022-03-27 00:05:37,312 [DEBUG]  rpc: getblockchaininfo []
2022-03-27 00:05:37,341 [INFO]  No timelocked coins in wallet, not announcing fidelity bond
2022-03-27 00:05:37,342 [ERROR]  Failed to create offers.
2022-03-27 00:05:37,342 [INFO]  Shutting down YieldGenerator service.
2022-03-27 00:05:37,940 [INFO]  joined: #joinmarket-pit-test ergo.test
2022-03-27 00:05:37,941 [INFO]  All IRC servers connected, starting execution.
[...]
2022-03-27 00:15:37,943 [INFO]  Failed to start after 10 minutes, giving up.
2022-03-27 00:15:37,948 [INFO]  Lost IRC connection to: ergo.test . Should reconnect automatically soon.
2022-03-27 00:15:37,949 [DEBUG]  IRC connection lost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionLost'>: Connection to the other side was lost in a non-clean fashion: Connection lost.
```

`jmwalletd` was started via `supervisord` which logs the following:
```
jm_regtest_joinmarket | 2022-03-27 00:15:38,166 INFO exited: jmwalletd (exit status 0; expected)
```

Do you think there is a good workaround or a fix that can be applied, or is this the "expected behaviour"?
I tried to pin it down and provide the minimal reproducible steps. I am happy to assist in any way if you need more information.

OK, I've reproduced the behaviour (as I understand your report):

If you try to start the maker service for an empty wallet using the API, it will return an error message: `{"message": "Maker could not start, no confirmed coins."}` as intended, and the yield generator service will not start and the `/status` output remains as expected (session for wallet acting but nothing running, i.e. `{"session": true, "maker_running": false, "coinjoin_in_process": false, "wallet_name": "regtest1.jmdat"}`).

When you do the same in the case that a wallet has coins, but only frozen coins you get the following log output daemon side:
```
2022-04-14 13:48:27,399 [ERROR]  do not have any coins left
2022-04-14 13:48:27,436 [INFO]  No timelocked coins in wallet, not announcing fidelity bond
2022-04-14 13:48:27,437 [ERROR]  Failed to create offers.
2022-04-14 13:48:27,438 [INFO]  Shutting down YieldGenerator service.
2022-04-14 13:48:28,025 [INFO]  joined: #joinmarket-pit-test ubuntu20-juno
2022-04-14 13:48:28,026 [INFO]  joined: #joinmarket-pit-test ubuntu20-juno
2022-04-14 13:48:28,027 [INFO]  Could not connect to *ALL* servers yet, waiting up to 60 more seconds.
2022-04-14 13:48:28,028 [INFO]  All message channels connected, starting execution.
```
indicating backend messaging started up as if the yield generator was actually starting. Obviously there is a logic error somewhere causing this difference between the cases, when they should be identical. Should be able to find it quite quickly I think (and, like the previous bug we looked at, I strongly suspect this is not the RPC-API but some bug internal. Well, let's see :) ).

--- a/jmclient/jmclient/wallet_rpc.py
+++ b/jmclient/jmclient/wallet_rpc.py
@@ -612,8 +612,9 @@ class JMWalletDaemon(Service):
                 # sync has already happened (this is different from CLI yg).
                 # note: an edge case of dusty amounts is lost here; it will get
                 # picked up by Maker.try_to_create_my_orders().
-                if not len(self.services["wallet"].get_balance_by_mixdepth(
-                    verbose=False, minconfs=1)) > 0:
+                gbbm = self.services["wallet"].get_balance_by_mixdepth(
+                    verbose=False, minconfs=1)
+                if len(gbbm) == 0 or all([v==0 for v in gbbm.values()]):
                     # note: this raise will prevent the setup
                     # of the service (and therefore the startup) from
                     # proceeding:

